client/signup/main.jsx

```diff
@@ -352,22 +330,12 @@ class Signup extends React.Component {
 			flow: this.props.flowName,
 		} );
 
-		if ( dependencies && ( dependencies.cartItem || dependencies.domainItem ) ) {
-			this.handleLogin( dependencies, destination );
-		} else {
-			this.setState( {
-				loginHandler: this.handleLogin.bind( this, dependencies, destination ),
-			} );
-		}
+		this.handleLogin( dependencies, destination );
```

Comment:
Are we confident it's safe to bypass the `dependencies` check?
Looking at `handleLogin` I see that there are a few attempts to access properties with dot notation with no prior guarding.
Maybe it'd be worth adding a guard clause to `handleLogin`, what do you think?

Reply:
The `dependencies` object comes from the `SignupFlowController.onComplete` handler which gets it with the [getSignupDependencyStore](https://github.com/Automattic/wp-calypso/blob/fb810840a4b0f73cca6fccd47244ead58dd0967e/client/state/signup/dependency-store/selectors.js#L11) Redux selector. All null-ish values are defaulted to an empty object there.

Also, before this patch, the `handleLogin` was bound to the unchecked `dependencies` value and assigned to `loginHandler`:
```js
loginHandler = handleLogin.bind( this, dependencies, destination );
// later in another module
loginHandler();
```
It would be crashing already if `dependencies` could be null-ish.

It's therefore safe to assume that `dependencies` is always a valid object.